graph/adjacency/impl/ThingAdjacencyImpl.java

```diff
@@ -66,13 +66,13 @@
     IID[] infixTails(ThingEdge edge) {
         if (edge.encoding().isOptimisation()) {
             if (direction.isOut()) {
-                return new IID[]{edge.outIID().infix().asRolePlayer().tail(), edge.to().iid().prefix(), edge.to().iid().type()};
+                return new IID[]{edge.outIID().infix().asRolePlayer().tail(), edge.toIID().prefix(), edge.toIID().type()};
```

Comment:
to avoid infinite circular calls to `to()` when NOT caching (which we don't need to here), we switch from using `to().iid()` to `toIID()`, which doesn't even have to ask the `ThingGraph` to do any conversion of iid to vertex